src/linker/Linker/Driver.cs

```diff
@@ -674,11 +674,20 @@ public bool Run (ILogger customLogger = null)
 
 			try {
 				p.Process (context);
+			} catch (Exception ex) {
+				Console.Error.WriteLine ("IL Linker exited with error:");
+				if (ex is LinkerFatalErrorException le) {
+					context.LogMessage (le.MessageContainer);
```

Comment:
Why log only `LinkerFatalErrorException` exceptions?

Reply:
I think the difference is that `LinkerFatalErrorException` has the `MessageContainer` so we can do better job reporting the error to MSBuild (as it will have a specific code, category and so on).
For the same reason we need to handle inner exceptions as well - although maybe we should handle them by producing an error to the log as well.

Reply:
That's right but we could create MessageContainer for all other exceptions from Message or ToString and log them too, right now the text goes to Console only.

We should also add check to LinkerFatalErrorException that the MessageContainer is of Error type.